### PR TITLE
feat: Add `vec_mul` function.

### PR DESCRIPTION
**Pull request by linyihai**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
Part of https://github.com/GreptimeTeam/greptimedb/issues/5120

## What's changed and what's your intention?

 VEC_MUL(vec0, vec1): Multiplies corresponding elements of two vectors, returns a vector.

## PR Checklist
Please convert it to a draft if some of the following conditions are not met.

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.
- [x] API changes are backward compatible.
- [x] Schema or data changes are backward compatible.

### CODE REVIEW

**Review by evenyag — COMMENTED** [review]
@zhongzc PTAL

File: src/common/function/src/scalars/vector/vector_mul.rs [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            Some("[7.0,8.0,9.0]".to_string()),
+            None,
+        ]));
+
+        let input1 = Arc::new(StringVector::from(vec![
+            Some("[1.0,2.0,3.0]".to_string()),
+            Some("[8.0,10.0,12.0]".to_string()),
+            Some("[7.0,8.0,9.0]".to_string()),
+            None,
+        ]));
```
[the remark this answers is not in the document]
Comment (linyihai) [reply src/common/function/src/scalars/vector/vector_mul.rs]: Thank you. I had updated the test.

**Review by zhongzc — APPROVED** [review]

**Review by evenyag — APPROVED** [review]
LGTM

**Comment by evenyag** [comment]
Looks like the CI will fail if the docs-required check box is checked.

**Pull request merged**